### PR TITLE
Add more API docs in Jupyter Book

### PR DESCRIPTION
**Pull request by rahulporuri**
This PR adds the following API docstrings in the Jupyter Book docs

- `extract_frames`
- `check_labels`
- `create_training_dataset`
- `train_network`

Additionally, the API file introduced in #1795 was renamed. The new naming scheme is hopefully easier to understand. The renaming was also necessary as the earlier naming scheme which involved the use of module name for filename caused issues where multiple functions were defined in the same module.

This is a continuation of #1795 .

### CODE REVIEW

**Comment by rahulporuri** [comment]
@AlexEMG and @MMathisLab - i'll finish this task next weekend, unless someone else beats me to it!

remaining functions are

- `label_frames`
- `evaluate_network`
- `analyze_videos`
- `extract_outlier_frames`
- `refine_labels` and `merge_datasets`

File: deeplabcut/generate_training_dataset/frame_extraction.py [hunk 1]
```diff
@@ -76,104 +76,179 @@ def extract_frames(
...
+        frames extracted. Otherwise one can select a subset by passing those paths.
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    Use the function ``add_new_video`` at any stage of the project to add new videos to
+    the config file and extract their frames.
```
Comment (rahulporuri) [inline deeplabcut/generate_training_dataset/frame_extraction.py]:
```suggestion
    Use the function ``add_new_videos`` at any stage of the project to add new videos
    to the config file and extract their frames.
```

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 2]
```diff
@@ -712,52 +722,97 @@ def create_training_dataset(
...
-        Type of augmenter. Currently default, imgaug, tensorpack, and deterministic are supported.
+        Returns None if training dataset could not be created.
 
-    posecfg_template: string (optional, default=None)
-        Path to a pose_cfg.yaml file to use as a template for generating the new one for the current iteration. Useful if you
-        would like to start with the same parameters a previous training iteration. None uses the default pose_cfg.yaml.
+    Notes
+    -----
+    Use the function ``add_new_video`` at any stage of the project to add more videos
+    to the project.
```
Comment (rahulporuri) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]:
```suggestion
    Use the function ``add_new_videos`` at any stage of the project to add more videos
    to the project.
```

**Review by MMathisLab — APPROVED** [review]

File: deeplabcut/generate_training_dataset/frame_extraction.py [hunk 3]
```diff
@@ -76,104 +76,179 @@ def extract_frames(
...
     videos_list=None,
 ):
-    """
-    Extracts frames from the videos in the config.yaml file. Only the videos in the config.yaml will be used to select the frames.\n
-    Use the function ``add_new_video`` at any stage of the project to add new videos to the config file and extract their frames.
+    """Extracts frames from the project videos.
 
-    The provided function either selects frames from the videos in a randomly and temporally uniformly distributed way (uniform), \n
-    by clustering based on visual appearance (k-means), or by manual selection.
+    Frames will be extracted from vidoes listed in the config.yaml file.
```
Comment (jeylau) [inline deeplabcut/generate_training_dataset/frame_extraction.py]: videos

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 4]
```diff
@@ -288,36 +288,46 @@ def check_labels(
     draw_skeleton=True,
     visualizeindividuals=True,
 ):
-    """
-    Double check if the labels were at correct locations and stored in a proper file format.\n
-    This creates a new subdirectory for each video under the 'labeled-data' and all the frames are plotted with the labels.\n
+    """Check the labelled frames.
```
Comment (jeylau) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: labeled

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 5]
```diff
@@ -288,36 +288,46 @@ def check_labels(
...
+    scale : float, default=1
         Change the relative size of the output images.
 
-    dpi : int, optional
-        Output resolution. 100 dpi by default.
+    dpi : int, optional, default=100.
+        Output resolution in dpi.
 
-    draw_skeleton: bool, default True.
+    draw_skeleton: bool, default: True.
```
Comment (jeylau) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: default=True
(consistent with the rest?)

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 6]
```diff
@@ -288,36 +288,46 @@ def check_labels(
...
+        List of at least 3 matplotlib markers. The first one will be used to indicate
+        the human ground truth location (Default: +)
 
-    scale : float, default =1
+    scale : float, default=1
         Change the relative size of the output images.
 
-    dpi : int, optional
-        Output resolution. 100 dpi by default.
+    dpi : int, optional, default=100.
```
Comment (jeylau) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: perhaps remove the final "."

**Review by jeylau — COMMENTED** [review]
Very minor typos @rahulporuri, otherwise awesome contribution 🤩 Thank you very much!

**Comment by rahulporuri** [comment]
Thanks for the comments @jeylau !

File: deeplabcut/generate_training_dataset/frame_extraction.py (hunk 3 above)
Comment (rahulporuri) [inline deeplabcut/generate_training_dataset/frame_extraction.py]:
```suggestion
    Frames will be extracted from videos listed in the config.yaml file.
```

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py (hunk 4 above)
Comment (rahulporuri) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]:
```suggestion
    """Check the labeled frames.
```

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py (hunk 6 above)
Comment (rahulporuri) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]:
```suggestion
    dpi : int, optional, default=100
```

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py (hunk 5 above)
Comment (rahulporuri) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]:
```suggestion
    draw_skeleton: bool, default=True
```

**Comment by rahulporuri** [comment]
@jeylau i've addressed your comments. thanks for the corrections. if you didn't know already, when you are providing comments, you can directly suggest changes using the `+-` button (see below). That's what I ended up doing to make the changes you recommended directly using the GitHub interface.

image

**Comment by jeylau** [comment]
Thanks for the tip! That button renders very poorly on safari, I hadn't seen it before 😅

**Review by jeylau — APPROVED** [review]

**Review by MMathisLab — APPROVED** [review]

**Pull request merged**